### PR TITLE
feat: Added social signin docs

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes:
This PR adds documentation for the `signInWithRedirect` API.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx [hunk 1]
```diff
@@ -247,193 +251,18 @@ const updatedAwsConfig = {
 Amplify.configure(updatedAwsConfig);
 ```
 
-### Full Sample
```
Comment (jimblanc) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:250]: Removed the full example as it wasn't adding much that wasn't covered in the other examples.

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx [hunk 2]
```diff
@@ -82,17 +82,24 @@ import all1 from "/src/fragments/lib/auth/common/social_signin_web_ui/configure_
...
 
+</InlineFilter>
+
 <InlineFilter filters={['js']}>
 
+After configuring the OAuth endpoints (Cognito Hosted UI), you can integrate your App by invoking the `signInWithRedirect` API. Passing `Amazon`, `Facebook`, `Google`, or `Apple` on the `provider` argument (e.g `signInWithRedirect({ provider: 'Amazon' })`) will bypass the Hosted UI and federate immediately with the social provider as shown in the below React example. If you are looking to add a custom state, you are able to do so by passing a string (e.g. `signInWithRedirect({ customState: 'xyz' })`) value and listening for the custom state via Hub.
```
Comment (israx) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:93]: > and listening for the custom state via Hub.

That's something that we still don't support.

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx [hunk 3]
```diff
@@ -104,11 +111,8 @@ function App() {
...
       switch (event) {
-        case "signIn":
-          setUser(data);
-          break;
-        case "signOut":
-          setUser(null);
+        case "signInWithRedirect":
+          getUser();
           break;
         case "customOAuthState":
```
Comment (israx) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:117]: still not supported

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx [hunk 4]
```diff
@@ -159,48 +164,47 @@ function App() {
...
           setCustomState(data);
```
Comment (israx) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:173]: no supported yet
```suggestion
```

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx (hunk 3 above)
Comment (israx) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:117]:
```suggestion
```

File: src/pages/lib-v1/auth/social/q/platform/[platform].mdx [hunk 5]
```diff
@@ -104,11 +111,8 @@ function App() {
...
```
Comment (israx) [inline src/pages/lib-v1/auth/social/q/platform/[platform].mdx:118]:
```suggestion
```

**Review by israx — COMMENTED** [review]
Looks pretty good. We only need to remove the `custom0AuthState` event that we still don't support

**Review by israx — APPROVED** [review]
lgtm

**Pull request merged**